### ISSUE TITLE
What is the training performance?

### ISSUE THREAD

**Issue by idiomic** [mchong6/GANsNRoses#5]
This is great work!

I've been looking at many similar models for a business problem. Yours looks most promising, but I've noticed that information about the training performance was not included in the paper. How well does it perform, and would you consider it "efficient" on hardware?

Thanks,
Tyler

**Comment by mchong6** [comment]
I trained it for about 5 days on a Quadro RTX 6000 with a batch size of 6. In general, this model is quite expensive, and someone has commented on slow CPU inference for videos. Otherwise, I am unsure of how to measure or quantify performance.

**Comment by idiomic** [comment]
Well, quality results come at a price, yes?

I wonder if the model could be extended to maintain pretraining data. This could make training for specific uses easier and increase creativity due to a larger base of "experience" to draw from. 

How does the number of input / examples needed very with problem difficulty?

**Comment by idiomic** [comment]
I'm considering it for generating mass produced images in specific artists styles, with images / sketches given for the "inspiration" of the art.

**Comment by mchong6** [comment]
Do you mean something like finetuning? I suspect it might work, but I would think you would need at least 1000 images of a particular style.

Depending on your use case, you can consider approaches such as https://github.com/justinpinkney/toonify which requires less data. The downside is you don't get the diversity found in GANsN'Roses.

**Comment by idiomic** [comment]
After studying a bit more, I suspect the 5-10 times increase in data efficiency from "Differentiable Augmentation
for Data-Efficient GAN Training" (https://arxiv.org/pdf/2006.10738.pdf), smoother loss and generalization increase from "DON’T LET YOUR DISCRIMINATOR BE FOOLED" (https://openreview.net/pdf?id=HJE6X305Fm), and finally pretraining on a larger dataset from all artists' work and fine-tuned on specific artist's work might increase data efficiency enough.

If you would like to find out more about the project and optionally earn a cut of the profit, please send me a message!

This is a really great work. The code is clean, unlike many research projects. Great results and clear presentation. Congratulations!

**Comment by mchong6** [comment]
Unfortunately, I have quite a lot on my plate now. I hope it works out for you!
